src/Elm/Parser/Layout.elm

```diff
@@ -81,20 +81,18 @@ positivelyIndented : Parser State ()
 positivelyIndented =
     Combine.Parser
         (\state ->
-            Core.getCol
-                |> Core.andThen
-                    (\column ->
-                        let
-                            expectedColumn : Int
-                            expectedColumn =
-                                State.currentIndent state
-                        in
-                        if expectedColumn < column then
+            Core.map
+                (\column ->
+                    \indent ->
+                        if indent < column then
                             Core.succeed ( state, () )
 
                         else
                             Core.problem "must be positively indented"
-                    )
+                )
+                Core.getCol
+                |= State.currentIndent
+                |> Core.andThen identity
```

Comment:
We could even extract the whole core part and use `Combine.fromCore` (if even necessary)

```elm
positivelyIndented : Parser state ()
positivelyIndented =
    Core.map
        (\column ->
            \indent ->
                if indent < column then
                    Core.succeed ()

                else
                    Core.problem "must be positively indented"
        )
        Core.getCol
        |= State.currentIndent
        |> Core.andThen identity
        |> Combine.fromCore
```

Reply:
Wouldn't my version be faster because we don't redefine the core parser inside a function (the `\state ->` lambda)?

Also, maybe I'm not following which extra map you're thinking of.

Reply:
in the current we do `Core.succeed ( state, () )`, in the `fromCore` we internally do a `Core.map (\val -> ( state, val ))`.

But might be that constructing this small parser is more expensive than predefining+map but I'm sceptical